jupyter_ai_router/router.py

```diff
@@ -157,10 +167,73 @@ def _route_message(self, room_id: str, message: Message) -> None:
 
         # Check if it's a slash command
         if first_word and first_word.startswith("/"):
-            self._notify_slash_cmd_observers(room_id, message)
+            # Extract command and create trimmed message
+            parts = message.body.split(None, 1)  # Split into max 2 parts
+            command = parts[0] if parts else ""
+            trimmed_body = parts[1] if len(parts) > 1 else ""
+            
+            # Create a copy of the message with trimmed body (command removed)
+            trimmed_message = replace(message, body=trimmed_body)
+            
+            # Remove forward slash from command for cleaner API
+            clean_command = command[1:] if command.startswith("/") else command
+            
+            # Route to slash command observers
+            self._notify_slash_cmd_observers(room_id, trimmed_message, command, clean_command)
         else:
             self._notify_msg_observers(room_id, message)
 
+    def _command_matches(self, command: str, pattern: str) -> bool:
```

Comment:
1. This can be made a plain function outside this class since it just checks if a word matches a regex pattern encoded as a string.
2. Since we don't know if the first parameter is a command before running this function, I recommend renaming `command` to `word` or `first_word`.
3. (nitpick) I find methods are more readable when they start with a verb, e.g. `word_matches_pattern()`.